pkg/roll/farm.go

```diff
@@ -374,7 +394,32 @@ func (rlf *Farm) validateRoll(update roll_fields.Update, logger logging.Logger)
 
 // Tries to delete the given RU every second until it succeeds
 func (rlf *Farm) mustDeleteRU(id roll_fields.ID, logger logging.Logger) {
-	for err := rlf.rls.Delete(id); err != nil; err = rlf.rls.Delete(id) {
+	ctx, cancelFunc := transaction.New(context.Background())
+	defer cancelFunc() // technically not necessary once Commit() succeeds
+	err := rlf.rls.Delete(ctx, id)
+	if err != nil {
+		// this error is really bad because we can't recover from it
```

Comment:
Can we retry this operation? It seems like there's nothing else on the context at this point, so we could create a new one and try the rollstore.Delete again.